shared/src/hooks/useAirtable.ts

```diff
@@ -0,0 +1,118 @@
+import Airtable from "airtable";
+import dotenv from "dotenv";
+import { useEffect, useMemo, useState } from "react";
+import useAssetPrice from "./useAssetPrice";
+
+export interface ScheduleItem {
+  strikePrice: number;
+  baseYield: number;
+  participationRate: number;
+  barrierPercentage: number;
+}
+
+dotenv.config();
+
+Airtable.configure({
+  endpointUrl: "https://api.airtable.com",
+  apiKey: process.env.REACT_APP_AIRTABLE_API_KEY,
+});
+
+const BASE_NAME = "Earn";
+
+const base = Airtable.base("appkUHzxJ1lehQTIt");
+
+export const useAirtable = () => {
+  const [schedules, setSchedules] = useState<ScheduleItem[]>();
```

Comment:
looks like we're only interested in the latest schedule, in that case we dont have to store the entire array of `schedules`